src/app/MessageDef/MessageDefHelper.cpp

```diff
@@ -37,26 +37,32 @@ namespace app {
 // this is used to run in signle thread for IM message debug purpose
 namespace {
 uint32_t gPrettyPrintingDepthLevel = 0;
-char gLineBuffer[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
-size_t gCurLineBufferSize = 0;
+size_t gCurLineBufferSize          = 0;
+
+std::vector<char> & GetLineBuffer()
+{
+    static std::vector<char> sLineBuffer(CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE);
```

Comment:
Please don't use STL containers in the code that can be run on constrained devices.

Reply:
Changed to unique_ptr.